src/output.rs

```diff
@@ -17,10 +18,14 @@ pub enum Argument {
 }
 
 impl FilterPrototype {
-    pub fn new(name: &str, arguments: Vec<Argument>) -> FilterPrototype {
+    pub fn new(name: &str,
+               arguments: Vec<Argument>,
+               named_arguments: Vec<(String, Argument)>)
+               -> FilterPrototype {
         FilterPrototype {
             name: name.to_owned(),
             arguments: arguments,
+            named_arguments: named_arguments,
```

Comment:
🤦‍♂️

> What is the plan for actually applying the named arguments?

I'm not sure what you mean by "applying". I guess we'll have to change the signature of the fn for the tags, introducing a breaking change :(. 

> What exactly are the semantics for named arguments that we need to support, like mixing positional and named?

At first I started by only supporting named args at the end of the filter, but then I had the brilliant idea of checking how Shopify/liquid does it. [As you can see](https://github.com/Shopify/liquid/blob/c582b86f16b64f5df890807603f57914099f3868/lib/liquid/variable.rb#L93-L99), named args and positional args can be mixed.

> How can we make this easy for filter writers?

I've been flip-flopping on this. At first I thought of using a `HashMap`. After all, keys are always strings. Then I figured Erlang just uses lists of tuples, which for low amount of args is probably more performant. Now I don't know. For ease of writing, I think `HashMap`.